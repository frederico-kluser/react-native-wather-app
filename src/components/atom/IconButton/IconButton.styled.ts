import styled from 'styled-components/native';
import {GLOBAL_PADDING, ICON_SIZE} from '../../../styles/theme';

export const DeleteContainer = styled.TouchableOpacity`
  position: absolute;
  bottom: 11%;
  left: 0px;
  width: 100%;
`;

export const DeleteMessage = styled.Text`
  left: ${GLOBAL_PADDING}px;
  text-align: center;
`;

export const DeleteIcon = styled.Image`
  position: absolute;
  bottom: 20px;
  left: 50%;
`;

export const CloseContainer = styled.TouchableOpacity`
  right: ${GLOBAL_PADDING}px;
  position: absolute;
  top: ${GLOBAL_PADDING}px;
  z-index: 2;
`;

export const FavoriteContainer = styled.TouchableOpacity`
  left: ${GLOBAL_PADDING}px;
  position: absolute;
  top: ${GLOBAL_PADDING}px;
  z-index: 2;
`;

export const OnlyIcon = styled.Image`
  height: ${ICON_SIZE}px;
  width: ${ICON_SIZE}px;
`;
