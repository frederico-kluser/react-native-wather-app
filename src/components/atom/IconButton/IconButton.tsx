import React from 'react';
import {
  closeIcon2,
  heartFullIcon,
  heartIcon,
  trashIcon,
} from '../../../styles/icons';
import {
  CloseContainer,
  DeleteContainer,
  DeleteIcon,
  DeleteMessage,
  FavoriteContainer,
  OnlyIcon,
} from './IconButton.styled';

export const DeleteButton = ({onPress}: any) => (
  <DeleteContainer onPress={onPress}>
    <DeleteIcon source={trashIcon} />
    <DeleteMessage>Deletar Item</DeleteMessage>
  </DeleteContainer>
);

export const CloseButton = ({onPress}: any) => (
  <CloseContainer onPress={onPress}>
    <OnlyIcon source={closeIcon2} />
  </CloseContainer>
);

export const FavoriteButton = ({onPress, favorite}: any) => (
  <FavoriteContainer onPress={onPress}>
    <OnlyIcon source={favorite ? heartFullIcon : heartIcon} />
  </FavoriteContainer>
);
