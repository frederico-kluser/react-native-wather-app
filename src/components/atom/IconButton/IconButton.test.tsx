import React from 'react';
import renderer from 'react-test-renderer';
import {CloseButton, DeleteButton} from './IconButton';

test('renders correctly delete button', () => {
  const tree = renderer.create(<DeleteButton />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders correctly close button', () => {
  const tree = renderer.create(<CloseButton />).toJSON();
  expect(tree).toMatchSnapshot();
});
