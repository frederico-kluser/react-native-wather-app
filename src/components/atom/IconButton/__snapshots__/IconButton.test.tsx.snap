// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly close button 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "position": "absolute",
      "right": 16,
      "top": 16,
      "zIndex": 2,
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/close2.png",
      }
    }
    style={
      Array [
        Object {
          "height": 32,
          "width": 32,
        },
      ]
    }
  />
</View>
`;

exports[`renders correctly delete button 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "bottom": "11%",
      "left": 0,
      "opacity": 1,
      "position": "absolute",
      "width": "100%",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/trash.png",
      }
    }
    style={
      Array [
        Object {
          "bottom": 20,
          "left": "50%",
          "position": "absolute",
        },
      ]
    }
  />
  <Text
    style={
      Array [
        Object {
          "left": 16,
          "textAlign": "center",
        },
      ]
    }
  >
    Deletar Item
  </Text>
</View>
`;
